template/CMakeLists.txt

```diff
@@ -1,56 +1,66 @@
-cmake_minimum_required(VERSION 3.10)
+# include some defines automatically made by qpm
+include(qpm_defines.cmake)
 
-# set the project name
-project(#{id}) #Modify this to your project name
+cmake_minimum_required(VERSION 3.21)
+project(${COMPILE_ID})
 
-# specify the C++ standard
+# c++ standard
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
-
-# NDK Magic.
-# Note: On Windows, Use WSL with NDK installed in WSL too. 
-# For CLion, you'll likely also need to provide `-DCMAKE_ANDROID_NDK=/path/to/ndk/` in CMake options.
-
-# Put path to NDK such as PATH//TO//NDK
-if (NOT DEFINED CMAKE_ANDROID_NDK)
-    if(DEFINED ENV{ANDROID_NDK_ROOT})
-        set(ANDROID_NDK_ROOT ENV{ANDROID_NDK_ROOT})
-    else()
-        file (STRINGS "ndkpath.txt" CMAKE_ANDROID_NDK)
-    endif()
-endif()
-
-set(CMAKE_SYSTEM_NAME Android)
-set(ANDROID_NATIVE_API_LEVEL 25)
-set(ANDROID_ABI arm64-v8a)
-set(CMAKE_ANDROID_ARCH_ABI arm64-v8a)
-set(ANDROID_TARGET_ARCH arm64-v8a)
-set(CMAKE_ANDROID_STL_TYPE c++_static)
-
-set(CMAKE_C_COMPILER_TARGET aarch64-none-linux-android24)
-set(CMAKE_CXX_COMPILER_TARGET aarch64-none-linux-android24)
-set(CMAKE_TOOLCHAIN_FILE ${CMAKE_ANDROID_NDK}/build/cmake/android.toolchain.cmake)
-
-message("Using NDK ${CMAKE_ANDROID_NDK}")
-
-# Source files
-
-include_directories(include)
-include_directories(extern)
-include_directories(shared)
-include_directories(extern/libil2cpp/il2cpp/libil2cpp)
-# Uncomment these if you use these libraries
-include_directories(extern/codegen/include) # uncomment if no use
-
-include_directories(${CMAKE_ANDROID_NDK})
-include_directories(.)
+set(CMAKE_CXX_STANDARD_REQUIRED 20)
 
-set(CMAKE_CXX_STANDARD 20)
+# define that stores the actual source directory
+set(SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+# compile options used
+add_compile_options(-frtti -fexceptions)
+add_compile_options(-O3)
+# compile definitions used
+add_compile_definitions(VERSION=\"${MOD_VERSION}\")
+add_compile_definitions(ID=\"${MOD_ID}\")
+add_compile_definitions(MOD_ID=\"${MOD_ID}\")
+add_compile_definitions(DEBUGB=1) # disable debug logs
```

Comment:
I'm assuming you copied this from Chroma's repo, since this a define only used by Chroma. I believe CMake/NDK does set this value themselves, so it may not be wise to force it upon everyone

Reply:
Yeah i got it from chroma, ill remove it now